lib/ansible/modules/cloud/azure/azure_rm_keyvault.py

```diff
@@ -0,0 +1,436 @@
+        self.location = None
+        self.sku = None
+        self.vault_uri = None
+        self.tenant_id = None
+        self.object_id = None
+        self.application_id = None
+        self.permissions = None
+        self.tags = None
+
+        super(AzureRMKeyVault, self).__init__(self.module_arg_spec,
+                                              supports_check_mode=True,
+                                              supports_tags=True)
+
+    def exec_module(self, **kwargs):
+
+        # create a new vault variable in case the 'try' doesn't find a vault
+        vault = None
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        self.results['check_mode'] = self.check_mode
+
+        # retrieve resource group to make sure it exists
+        resource_group = self.get_resource_group(self.resource_group)
+        if not self.location:
+            # Set default location
+            self.location = resource_group.location
+
+        changed = False
+        results = dict()
+
+        try:
+            self.log('Fetching Vault {0}'.format(self.name))
+            vault = self.keyvault_client.vaults.get(self.resource_group, self.name)
+
+            # serialize object into a dictionary
+            results = vault.as_dict()
+
+            # don't change anything if creating an existing vault (unless outdated tags), but change if deleting it
+            if self.state == 'present':
+                changed = False
+
+                update_tags, results['tags'] = self.update_tags(results['tags'])
+                if update_tags:
+                    changed = True
+
+            elif self.state == 'absent':
+                changed = True
+
+        except CloudError:
+            # the vault does not exist (it's deleted) so create it
+            if self.state == 'present':
+                changed = True
+
+            else:
+                # you can't delete what is not there
+                changed = False
+
+        self.results['changed'] = changed
+        self.results['state'] = results
+
+        if self.tenant_id is None:
+            self.tenant_id = self.credentials['tenant']
+        if self.object_id is None:
+            self.object_id = self.credentials['client_id']
+
+        # return the results if your only gathering information
+        if self.check_mode:
+            return self.results
+
+        if changed:
+            if self.state == 'present':
+                # if you want to create or update a key vault
+                if not vault:
+                    # create new vault
+                    self.log('Creating vault {0}'.format(self.name))
+                    use_sku = Sku(self.sku)
+                    permission_preset = create_permissions(self)
+                    access_policy_lst = [AccessPolicyEntry(self.tenant_id, self.object_id, permission_preset, application_id=self.application_id)]
```

Comment:
Seems this is a list of `AccessPolicyEntry`, how about I want to have more than one `AccessPolicyEntry`

Reply:
This is granting access rights to current user/spn only.

Reply:
But the SDK accept a list, that means the service accept more than one accessPolicyEntries. How can we handle it?

Reply:
not going to include